packages/typespec-ts/test/integration/clientRequestIdHeader.spec.ts

```diff
@@ -0,0 +1,48 @@
+import { assert } from "chai";
+import ClientRequestIdClientFactory, {
+  ClientRequestIdClient
+} from "./generated/headers/clientRequestId/src/index.js";
+describe("ClientRequestIdClient", () => {
+  let client: ClientRequestIdClient;
+
+  beforeEach(() => {
+    client = ClientRequestIdClientFactory({
+      allowInsecureConnection: true,
+      retryOptions: {
+        maxRetries: 0
+      }
+    });
+  });
+
+  it("should append client-request-id header and value if no any input", async () => {
+    try {
+      const result = await client
+        .path("/special-headers/client-request-id")
+        .get();
+      assert.isNotNull(result.request.headers.get("client-request-id"));
+      assert.strictEqual(result.status, "204");
+    } catch (err) {
+      assert.fail(err as string);
+    }
+  });
+
+  it("should set client-request-id header and value if input is provided", async () => {
+    try {
+      const overrideReqId = "86aede1f-96fa-4e7f-b1e1-bf8a947cb804";
+      const result = await client
+        .path("/special-headers/client-request-id")
+        .get({
+          headers: {
+            "client-request-id": overrideReqId
```

Comment:
Is this feature not trying to allow customization of the header name that is used as client request id ? This is just pass a value to client-request-id, I don't see any customization of the header name ?

Reply:
If you really want to test that you are using the customized header name, the place you should pass the new header name is where you create the client.

Reply:
The customization is done in client factory side, https://github.com/Azure/autorest.typescript/pull/1928/files/47be9347ea3ec92bed16bccc42cbaa7035f86062#diff-f8f59fb0ca54835e863a4d0d428934e6d47ce3338fad4eeaaf191a089fa34b63R30-R35

Reply:
The test case is because our current implementation keeps their header as a parameter so customer may leverage their own request id to override our default ones. So in this testing I'd like to verify the overriding works.

Reply:
Maybe add a test that we can customize the header name not the predefined one from clientoptions and then assert result.request has this new customized name

Reply:
Yes, sure we could add that test case.

Reply:
https://github.com/Azure/autorest.typescript/pull/1928/files#diff-d6ec5b3bce965c3900cd5096805dc992c8fad53a46963c77400a948eba854bf7R49-R69